sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -450,8 +453,11 @@ IterableStream<ServiceBusReceivedMessage> peekMessages(int maxMessages, long seq
             sessionId).timeout(operationTimeout);
 
         final Flux<ServiceBusReceivedMessage> tracedMessages = tracer.traceSyncReceive("ServiceBus.peekMessages", messages);
-        // Subscribe to message flux so we can kick off this operation, but not to tracing - caller subscriber will take care of it.
-        messages.subscribe();
+
+        // Subscribe to message flux so we can kick off this operation
+        // this works because all messages come in batched messages from management node, which is deserialized as a list and
+        // then returned as Flux.fromIterable.
+        tracedMessages.subscribe();
```

Comment:
After we merge this pr, sometime later, it would be nice to find out why we originally did this "messages.subscribe()" eager subscription. IIUC, this subscribe call 'schedules' the work to the reactor thread, and while that work is pending or in progress, the flux is wrapped to IterableStream and returned. Iteration on that stream can also make a subscribe() call, leading to duplicate work.

Reply:
Sounds good, created #34680 so I don't forget